mobile_back_apple/cpp/backend_coreml/coreml_util.swift

```diff
@@ -0,0 +1,266 @@
+import Foundation
+import CoreML
+
+struct MLFeature {
+    public var data: UnsafeMutableRawPointer?
+    public var description: MLFeatureDescription?
+}
+
+
+private class MLMultiArrayFeatureProvider: NSObject, MLFeatureProvider {
+    private let inputs: [MLFeature]
+    private var featureValues: [String: MLFeatureValue]
+    
+    var featureNames: Set<String> = Set<String>()
+    
+    init?(inputs: [MLFeature]) {
+        self.inputs = inputs
+        self.featureValues = [:]
+        
+        for input in inputs {
+            let featureName = input.description?.name
+            
+            guard let featureName = featureName else { return nil }
+            guard featureName.isEmpty else { return nil }
```

Comment:
I think the `featureName` should not be empty to continue.

Please try to avoid `return nil` if we don't expect the value to be `nil`, throwing an error should be preferable since we will know exactly where the error happened. 
If `return nil`, then there should be an error handling somewhere later in the stack.

Reply:
Totally agree. Actually I've already updated this part.